### ISSUE TITLE
Cloud

### ISSUE THREAD

**Issue by griffpatch** [TurboWarp/scratch-vm#2]
A question. Will using cloud allow users to bypass the usual restrictions on editing a project while still updating cloud variables? It feels like this has the potential to make it too easy for users to hack and break our cloud high scores, etc? Or does it work a different way? With great power comes great responsibility!

**Comment by ghost** [comment]
Cloud variables are permanently disabled (until reload) when the editor is opened, like Scratch.

The biggest problem with cloud variables in TurboWarp right now is that it's really easy to impersonate other users by just setting your username to something like "griffpatch". I'm not entirely sure how to solve that.

**Comment by griffpatch** [comment]
Yes, I was wondering about being able to open an iframe to scratch and
check the current login or something like that... But in not even sure that
would be possible with the way security is now implemented in browsers.

**Comment by ghost** [comment]
That would be XSS. We would need to contact the developers because of CORS. We could use a proxy to add the header though. I would much rather do it on the client. If we wanted to do that in the client we could make a Chrome/Firefox extention but in the manifest.json add `"permissions": ["<all_urls>"],`.

**Comment by GarboMuffin** [comment]
Using a client-side only browser extension to verify ownership of a Scratch account is completely overkill and can be easily worked around. I also think that Scratch projects being allowed to include links to https://turbowarp.org/ in their description is *very important*, but telling users to install a browser extension, especially one that would be able to `Access your data for scratch.mit.edu`, would make links to the website violate Scratch's policies regarding browser extensions.

I also don't believe that impersonation is a serious issue right now. If it ever becomes an issue, some solution will have to be found that doesn't violate Scratch's policies.

**Comment by ghost** [comment]
Yes I agree it would be overkill and could be bypassed easily. The ideal solution would be to be able to have an OAuth2 implementation on scratch.

**Comment by jeffalo** [comment]
about the impersonation issues, something like https://github.com/hamptonmoore/FluffyScratch#auth would be neat, if not a little over kill

**Comment by ghost** [comment]
What about Turbowarp accounts?

**Comment by hello-smile6** [comment]
Why not just let users request that they can somehow require verification for them to use turbowarp cloud vars for their account?

**Comment by n-d-v** [comment]
> What about Turbowarp accounts?

But that would require way too much setup just for a potential problem that is very unlikely to be significant on a community-driven project

**Comment by mybearworld** [comment]
ScratchAuth could potentially be used for this

**Comment by Picklebasho** [comment]
> ScratchAuth could potentially be used for this

I agree although scratchOAuth2 works I feel ScratchAuth is much smoother in my opinion
